### PR TITLE
download_zenodo(): more features

### PR DESCRIPTION
**Pull request by florisvdh**
New message is now: `Will download 11 files (total size: 1.4 GiB) from DOI: .........`

### CODE REVIEW

**Review by ThierryO — COMMENTED** [review]
How strong are the benefits of this new feature compared to the cost of having yet another dependency?

**Comment by florisvdh** [comment]
The benefit of this information is not neglible for larger datasets.

An alternative could be to copy over the function code of `humanize` into `inborutils`, but that seems not so elegant. The `humanize` package itself is small. Maybe there are alternative packages.

**Comment by florisvdh** [comment]
@ThierryO , FWIW I've just been looking for alternative functions (present on my own installation :-)). Found no 'common package' dependency for this, though - even `units` has no function to automatically return human-readable output. Is depending on `R.utils` package more convenient?

```
> R.utils::hsize(78945)
[1] "77.1 KiB"
> gdata::humanReadable(78945)
[1] "77.1 KiB"
> humanize::natural_size(78945, suffix_type="binary")
77.1 KiB
> prettyunits::pretty_bytes(78945)
[1] "78.94 kB"
> fs::fs_bytes(78945)
77.1K> 
```

Only the first three examples are able to present the _binary_ (as opposed to decimal) representation (which I believe is the common standard to be taken) as a _character_.

**Comment by florisvdh** [comment]
To make the latter ('character') clear for the less obvious cases:

```
> class(humanize::natural_size(78945))
[1] "glue"      "character"
> class(fs::fs_bytes(78945))
[1] "fs_bytes" "numeric" 
```
Also, it can be seen that `fs::fs_bytes()` does not use the IEC symbol of the [kibibyte](https://en.wikipedia.org/wiki/Kibibyte).

**Comment by florisvdh** [comment]
@hansvancalster @ThierryO modifications and extras:

- dropped `humanize` dependency; now uses internal function `human_filesize()`
- tweaked messages
- added argument `quiet` to suppress messages (default: `FALSE`)

File: R/download_zenodo.R [hunk 1]
```diff
@@ -48,7 +50,8 @@
 #' }
 download_zenodo <- function(doi,
                             path = ".",
-                            parallel = FALSE) {
+                            parallel = FALSE,
+                            quiet = FALSE) {
 
   assert_that(is.string(doi), is.string(path))
   assert_that(is.flag(parallel))
```
Comment (hansvancalster) [inline R/download_zenodo.R]:
```suggestion
  assert_that(is.flag(parallel), is.flag(quiet))
```

**Review by hansvancalster — COMMENTED** [review]
Looks OK to me, just one extra assert_that needed. 
Only looked at the code, not tested.

File: R/download_zenodo.R (hunk 1 above)
Comment (florisvdh) [reply R/download_zenodo.R]: thanks :+1:

**Review by hansvancalster — APPROVED** [review]
OK for me, but will await @ThierryO review before merging.

File: R/download_zenodo.R [hunk 2]
```diff
@@ -48,10 +51,11 @@
 #' }
 download_zenodo <- function(doi,
                             path = ".",
-                            parallel = FALSE) {
+                            parallel = FALSE,
+                            quiet = FALSE) {
 
   assert_that(is.string(doi), is.string(path))
-  assert_that(is.flag(parallel))
+  assert_that(is.flag(parallel), is.flag(quiet))
```
Comment (ThierryO) [inline R/download_zenodo.R]:
```suggestion
  assert_that(is.flag(parallel), noNA(parallel), is.flag(quiet), noNA(quiet))
```

Note that `NA` is a 'flag' (a logical of length 1).

Comment (ThierryO) [reply R/download_zenodo.R]: You should import `noNA` too.

File: R/download_zenodo.R [hunk 3]
```diff
@@ -63,6 +67,10 @@ download_zenodo <- function(doi,
   req <- curl_fetch_memory(paste0(base_url, record))
   content <- fromJSON(rawToChar(req$content))
 
+  # Calculate total file size
+  totalsize <- sum(content$files$size) %>%
+                human_filesize
```
Comment (ThierryO) [inline R/download_zenodo.R]:
```suggestion
                human_filesize()
```
This makes it more clear that is is a function

File: R/download_zenodo.R [hunk 4]
```diff
@@ -144,3 +158,41 @@ download_zenodo <- function(doi,
...
+#' @param x An integer, i.e. the number of bytes (B).
+#'
+#' @return A string.
+#' @importFrom assertthat
+#' assert_that
+#' is.number
+#' @importFrom dplyr
+#' %>%
+#' @keywords internal
+human_filesize <- function(x) {
```
Comment (ThierryO) [inline R/download_zenodo.R:193]: Wouldn't it be useful to vectorise this function?

**Review by ThierryO — COMMENTED** [review]
Overall this looks good. Just a few minor comments.

File: R/download_zenodo.R (hunk 4 above)
Comment (florisvdh) [reply R/download_zenodo.R:193]: Well, thought about it, but didn't do it - for now it's just an internal function with a single purpose. Would you export it? If so, then it's more worth it to broaden its use.

File: R/download_zenodo.R (hunk 2 above)
Comment (florisvdh) [reply R/download_zenodo.R]: Great tip, thanks.

File: R/download_zenodo.R (hunk 3 above)
Comment (florisvdh) [reply R/download_zenodo.R]: True.

**Comment by florisvdh** [comment]
Thanks @ThierryO for the extra suggestions. Will look at remaining topics next week!

File: R/download_zenodo.R (hunk 4 above)
Comment (florisvdh) [reply R/download_zenodo.R:193]: Vectorised in bb248cd
Exported in 2f622fb

**Pull request merged**